### DISCUSSION TITLE
Question about Python 3.10+ support in BentoML (>=1.3.10) and its dependency simple_di

### DISCUSSION THREAD

**Discussion by oogou11** [bentoml/BentoML#5140] [Q&A] [answered]
I’m using the BentoML framework (version >= 1.3.10) and noticed that it explicitly states support for Python 3.10, 3.11, and 3.12. However, one of its dependencies, `simple_di`, seems to only declare support for Python 3.6 through 3.9 in its PyPI page and `setup.py` file.

Here are my questions:  

1. Has `simple_di` been tested and confirmed to work with Python 3.10+?  
2. Is it possible that the Python version range in `setup.py` hasn’t been updated, leading to this apparent mismatch with BentoML’s stated compatibility?  
3. If `simple_di` hasn’t been fully validated for Python 3.10+, could there be potential compatibility issues?

Thank you for any insights or clarifications! I’d be happy to help with testing if needed.

**Comment by frostming** [comment, marked as answer]
1. `simple-di` doesn't cover Python 3.10+ in its own CI, but since it is a home-grown dependency used by BentoML, the latter is tested against Python 3.10+
2. I don't think it needs to be updated, the classifiers is an info-only metadata and what is important is the [`python_requires` value](https://github.com/bentoml/simple_di/blob/aa982934b0c4144f30efb1d87302bac0522d3715/setup.py#L27).
3. Same as 1

**Reply by oogou11** [reply]
Thank you for your clarification.

**Comment by aarnphm** [comment]
quick chime in here, fwiw we should add test matrix to 3.10+, but from our end it doesn't seem like anything breaks for 3.11 and beyond.

**Reply by oogou11** [reply]
Thank you for the reply and support.
